docs/remote/containers.md

```diff
@@ -231,6 +235,51 @@ Selecting the **Remote-Containers: Add Development Container Configuration Files
 
 To learn more about creating `devcontainer.json` files, see [Create a Development Container](/docs/remote/create-dev-container.md).
 
+## Dev container features
+
+Dev container features provide a smooth path for customizing your container definitions.
+
+When you use **Remote-Containers: Add Development Container Configuration Files**, you're presented a list of scripts to customize the existing dev container configurations, such as installing Git or the Azure CLI:
+
+![Dev container features in Command Palette](images/containers/container-features.png)
+
+When you rebuild and reopen in your container, the features you selected will be available in your `devcontainer.json`:
+
+```json
+"features": {
+		"github-cli": "latest"
+	}
+```
+
+You'll get IntelliSense when editing the `"features"` property in the `devcontainer.json` directly:
+
+![Intellisense when modifying terraform feature](images/containers/features-intellisense.png)
+
+The features are sourced from the [script library](https://github.com/microsoft/vscode-dev-containers/tree/main/script-library/docs) in the vscode-dev-containers repo.
+
+The **Remote-Containers: Configure Container Features** command allows you to update an existing configuration.
+
+    > **Note:** Features support for GitHub Codespaces is coming soon.
+
+## Pre-building dev container images
+
+You can pre-build images with the tools you need rather than newly creating and building images for each project. Using pre-built images can be simpler and allows you to pin to a specific version of tools, avoiding potential breaks. Pre-building is especially valuable in CI processes.
+
+You can use the [devcontainer CLI](/remote/devcontainer-cli.md) to facilitate pre-builds.
+
+The process could be as follows:
+* [Create](/docs/editor/versioncontrol.md/#_initialize-a-repository) a source code repository
+* Create a dev container configuration, customizing as you wish (including [features](#Dev-container-features))
+* Use the [`devcontainer CLI`](/docs/remote/devcontainer-cli.md) to build your image (the `devcontainer CLI` supports building images with features)
+* [Push](https://docs.docker.com/engine/reference/commandline/push/) your image. You can then change the dev container in your source repository to reference the image directly.
+
+```
+$ devcontainer build --image-name your-registry.azurecr.io/your-image-name
+$ docker push your-registry.azurecr.io/your-image-name
+```
+
+You can push your image to a container registry, like the [Azure Container Registry](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-docker-cli?tabs=azure-cli) or [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#pushing-container-images).
```

Comment:
Maybe we just call out Docker again here? I know we put that in the push step but if I'm skimming through the steps, I may not click on the Push link and so miss that we highlighted docker there. I think it makes sense to just list Docker in the list of registries here again.

Reply:
Makes sense, added mention of Docker Hub.